winforms/src/toga_winforms/app.py

```diff
@@ -32,9 +34,43 @@ def winforms_FormClosing(self, sender, event):
                 event.Cancel = True
 
 
-class App:
+class App(Scalable):
     _MAIN_WINDOW_CLASS = MainWindow
 
+    # ------------------- Set the DPI Awareness mode for the process -------------------
+    # This needs to be done at the earliest and doing this in __init__() or
+    # in create() doesn't work
```

Comment:
Can you identify what happens between the import and the `__init__` which prevents it from working? I don't think there should be any UI functions called during that time.

Reply:
The following is the initial starting calls for the test script in this PR, when using the latest main branch:
```
In Scalable Class
In Widget Class
In Container Class
In Window Class
In MainWindow Class
In App Class
In __init__() method of App Class
In main_loop method of App Class
In run_app method of App Class
In create method of App Class
...
```
I have attached the full file: [helloworld lifecycle.txt](https://github.com/beeware/toga/files/13043581/helloworld.lifecycle.txt)

As we can see, other UI Classes are accessed before App Class. But none of their __init__() methods are called before create method of App Class. So there should be no problem for the DPI Awareness setting to work when it is in __init__() method of App class. But for some reason, the setting of DPI Awareness fails, when done in either __init__() or create() method. 

Also, Microsoft recommends to not set DPI Awareness via API and instead to set via manifest: https://learn.microsoft.com/en-us/windows/win32/hidpi/setting-the-default-dpi-awareness-for-a-process#:~:text=We%20recommended%20that%20you%20specify%20the%20default%20process%20DPI%20awareness%20via%20a%20manifest%20setting.%20While%20specifying%20the%20default%20via%20API%20is%20supported%2C%20it%20is%20not%20recommended 

So, the API based DPI Awareness setting calling order is maybe more fragile.

Reply:
My best guess is that a Hwnd is being created somewhere before the calling of the __init__() method of App Class, which leads to the failing of call to set DPI Awareness.

Reply:
Unfortunately doing it via manifest isn't an option for Toga, because it has to work within any Python process.

Reply:
There is also: SetHighDpiMode https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.application.sethighdpimode. But it does not work with python.

Reply:
Why, what happens?

I don't think we could use that function anyway, because the "applies to" section doesn't include any ".NET Framework" versions, which are the ones that are bundled with Windows.

Reply:
It will complain that Application has no such method. 
Also, I think because we are not setting it via this function, hence the events like OnDpiChanged are not being triggered.

Reply:
That can't be the reason, because SetHighDpiMode isn't included in .NET Framework, but it says DpiChanged and the other events are supported by .NET Framework 4.7 and later, both [here](https://learn.microsoft.com/en-us/dotnet/desktop/winforms/high-dpi-support-in-windows-forms?view=netframeworkdesktop-4.8#new-dpi-change-events) and [here](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.form.dpichanged?view=windowsdesktop-7.0).

Reply:
See https://github.com/beeware/toga/pull/2155#discussion_r1385391970